oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocument.java

```diff
@@ -81,7 +79,7 @@ void addSpellcheck(String value) {
     // ref: https://www.elastic.co/blog/strings-are-dead-long-live-strings
     // (interpretation of date etc: https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html)
     void addProperty(String fieldName, Object value) {
-        properties.computeIfAbsent(fieldName, s -> new ArrayList<>()).add(value);
+        properties.computeIfAbsent(fieldName, s -> new LinkedHashSet<>()).add(value);
```

Comment:
Any specific reason to use LinkedHashSet?

Reply:
two reasons:
- remove duplicates
- keep the insert order

Reply:
Do we need order here?

Reply:
is not essential but it's good to keep the original order. Elastic stores this data in an ordered list. For each term, the position is stored as well. In general, searching for two terms that are closer has a higher score than two terms that are far apart.